src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.Map.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System.Buffers.Binary;
+
+namespace System.Security.Cryptography.Encoding.Tests.Cbor
+{
+    internal partial class CborReader
+    {
+        public ulong? ReadStartMap()
+        {
+            CborInitialByte header = PeekInitialByte(expectedType: CborMajorType.Map);
+            ulong arrayLength = checked((ulong)ReadUnsignedInteger(header, out int additionalBytes));
+            AdvanceBuffer(1 + additionalBytes);
+            _remainingDataItems--;
+
+            PushDataItem(CborMajorType.Map, 2 * arrayLength);
```

Comment:
Can this overflow? If so, either a checked cast or a value test (exceeding max/2) for an exception better than OverflowException.

Reply:
> an exception better than OverflowException.

Which exception type would you recommend here? I'm using OverflowException in similar error modes elsewhere.

Reply:
If similar cases are already using that exception type, then OverflowException is consistent.  By "better" I just meant "more related to the feature area"; like if things were an InvalidCborException, or whatever.